src/Microsoft.OData.Client/Materialization/ODataLoadNavigationPropertyMaterializer.cs

```diff
@@ -99,12 +102,11 @@ protected override bool ReadImplementation()
                         this.MaterializeEntryPlan,
                         this.responseInfo.IsContinuation);
                 }
-                else
+                else if (property.EdmProperty.Type.TypeKind() == EdmTypeKind.Entity)

```

Comment:
In which block do complex properties and/or complex collection properties get handled?

Reply:
@habbes This block https://github.com/OData/odata.net/blob/4e263ca4b94d01f2d105d8282c62ec272cfcd1bb/src/Microsoft.OData.Client/LoadPropertyResult.cs#L220